common/frame_session.go

```diff
@@ -749,6 +749,33 @@ func (fs *FrameSession) onFrameNavigated(frame *cdp.Frame, initial bool) {
 			fs.ctx, fs.targetID.String(), frame.URL,
 			trace.WithAttributes(attribute.String("url", frame.URL)),
 		)
+
+		var (
+			spanID   = fs.mainFrameSpan.SpanContext().SpanID().String()
+			newFrame = fs.manager.getFrameByID(frame.ID)
+		)
+		// Set k6SpanId property in the page so it can be retrieved when pushing
+		// the Web Vitals events from the page execution context and used to
+		// correlate them with the navigation span to which they belong to.
+		if newFrame != nil {
```

Comment:
This `if isMainFrame := frame.ParentID...-> if newFrame -> go func -> if err != nil` is too nested, making it harder to follow. Can we find a way to make it flat?

Reply:
It is a little bit nested, but there are different set of actions depending on the different conditions. In this case I think being nested makes it more explicit and easier to understand, I believe making it flatter would make it more difficult to understand. Can you see a way on how to improve it?

Reply:
I think nesting creates cognitive pressure, but no strong opinions. Sure, there's a similar code [here](https://github.com/grafana/xk6-browser/blob/86329fbb36f04c962073298c25233553a952e4f9/common/network_manager.go#L377-L411).